book/src/proposals/booting-from-snapshot.md

```diff
@@ -0,0 +1,262 @@
+# Booting From a Snapshot
+
+## Problem
+
+A validator booting from a snapshot should confirm that they are caught up to
+the correct cluster before voting. There are two major risks if they do not:
+
+### Wrong Cluster
+
+The validator may be fed a snapshot and gossip info for a different (though not
+necessarily malicious) cluster. If that cluster is at slot 100,000 while the
+cluster the validator intended to join is at 10,000, if the validator votes
+once on the wrong cluster, it must not vote on the correct cluster for at least
+90,000 slots to avoid being slashed. This is because slashing is based on slot
+numbers, so when the validator makes a vote `V` on the wrong cluster's slot
+100,000, it is locked out of voting on any slot < `100,000 + lockout(V)` on the
+correct cluster.
+
+### Voting Before Catching Up
+
+Any vote that a validator signs must have its lockout observed. If a validator
+has to reboot, the most secure way for it to check if its lockouts apply to the
+new set of forks is to see the votes on the ledger. If it can do this, it knows
+that those votes do not lock it out of voting on the current fork, so it just
+needs to wait for the lockouts of the votes not on the ledger to expire.
+However, if the validator votes before catching up, the votes will not go onto
+the ledger, so if the validator reboots, it will have to assume that the votes
+lock it out from voting again.
+
+
+## Snapshot Verification Overview
+
+A booting validator needs to have some set of trusted validators whose votes it
+will rely upon to ensure that it has caught up to the cluster and has a valid
+snapshot. With that set, the validator can follow the procedure detailed in the
+`Boot Procedure` section.
+
+For the sections below:
+
+Assume an arbitrary tower state `L`, a snapshot root `S`, and a trusted bank
```

Comment:
Can we be more liberal with the variable names.. Like tower state could be `TS` at least or even `TowerState`, snapshot root `SR`, trusted bank slot `TBS`.

Reply:
And why not lowercase?  Pretty weird to see the capital letters passed into what looks like Rust functions.

Reply:
lower case is great too

Reply:
The new variable names add a lot of verbosity. Some things are definitely clearer, but I'm not sure if I like it in all cases...